src/app_service/service/settings/service.nim

```diff
@@ -985,3 +985,119 @@ QtObject:
     except ValueError:
       error "parse lastTokensUpdate: ", lastTokensUpdate
     return self.settings.lastTokensUpdate
+
+  ### News Feed Settings ###
+  proc notifSettingStatusNewsChanged*(self: Service) {.signal.}
+
+  proc toggleNewsFeedEnabled*(self: Service, value: bool): bool =
+    try:
+      let response = status_settings.toggleNewsFeedEnabled(value)
+      if not response.error.isNil:
+        raise newException(RpcException, response.error.message)
+
+      self.settings.newsFeedEnabled = value
+      self.notifSettingStatusNewsChanged()
+      return true
+    except Exception as e:
+      error "error: ", procName="toggleNewsFeedEnabled", errName = e.name, errDesription = e.msg
+      return false
+
+  proc getNotifSettingStatusNews*(self: Service): string {.slot.} =
+    result = VALUE_NOTIF_SEND_ALERTS # Default value
+
+    var newsFeedEnabled = false
+    var newsNotificationsEnabled = false 
+
+    if self.initialized:
+      newsFeedEnabled = self.settings.newsFeedEnabled
+      newsNotificationsEnabled = self.settings.newsNotificationsEnabled
+    else:
+      try:
+        var response = status_settings.newsFeedEnabled()
+        if not response.error.isNil:
+          error "error reading news feed enabled setting: ", errDescription = response.error.message
+          return
+        newsFeedEnabled = response.result.getBool
+
+        response = status_settings.newsNotificationsEnabled()
+        if not response.error.isNil:
+          error "error reading news notifications enabled setting: ", errDescription = response.error.message
+          return
+        newsNotificationsEnabled = response.result.getBool
+      except Exception as e:
+        let errDesription = e.msg
+        error "reading news settings error: ", errDesription
+        return
+
+    # We convert the bools to the right setting
+    # Send alerts means the News Feed is enabled + notifications are enabled
+    # Deliver quietly means the News Feed is enabled + notifications are disabled
+    # Turn off means the News Feed is disabled
```

Comment:
Just to clarify.. since it confused me when I was reading!

```suggestion
    # Turn off means the News Feed is disabled + notifications are disabled
```